core/domain/state_domain.py

```diff
@@ -1120,9 +1120,17 @@ def _update_content_ids_in_assets(self, old_ids_list, new_ids_list):
         content_ids_to_delete = set(old_ids_list) - set(new_ids_list)
         content_ids_to_add = set(new_ids_list) - set(old_ids_list)
         for content_id in content_ids_to_delete:
-            self.content_ids_to_audio_translations.pop(content_id)
+            if not content_id in self.content_ids_to_audio_translations.keys():
```

Comment:
Here and below, do you actually need the ".keys()" part?

Reply:
Not required, removed!